<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML+RDFa 1.0//EN"
  "http://www.w3.org/MarkUp/DTD/xhtml-rdfa-1.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" version="XHTML+RDFa 1.0" dir="ltr"
  xmlns:content="http://purl.org/rss/1.0/modules/content/"
  xmlns:dc="http://purl.org/dc/terms/"
  xmlns:foaf="http://xmlns.com/foaf/0.1/"
  xmlns:og="http://ogp.me/ns#"
  xmlns:rdfs="http://www.w3.org/2000/01/rdf-schema#"
  xmlns:sioc="http://rdfs.org/sioc/ns#"
  xmlns:sioct="http://rdfs.org/sioc/types#"
  xmlns:skos="http://www.w3.org/2004/02/skos/core#"
  xmlns:xsd="http://www.w3.org/2001/XMLSchema#">

<head profile="http://www.w3.org/1999/xhtml/vocab">
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<link rel="shortcut icon" href="BEAST_favicon.ico" type="image/vnd.microsoft.icon" />
<link rel="shortlink" href="52.html" />
<link rel="canonical" href="discrete-phylogeographic-analysis.html" />
<meta name="Generator" content="Drupal 7 (http://drupal.org)" />
  <title>Discrete Phylogeographic Analysis | BEAST</title>
  <style type="text/css" media="all">
@import url("system.base.css%3Fou07bk.css");
@import url("system.menus.css%3Fou07bk.css");
@import url("system.messages.css%3Fou07bk.css");
@import url("system.theme.css%3Fou07bk.css");
</style>
<style type="text/css" media="all">
@import url("field.css%3Fou07bk.css");
@import url("node.css%3Fou07bk.css");
@import url("search.css%3Fou07bk.css");
@import url("user.css%3Fou07bk.css");
@import url("ckeditor.css%3Fou07bk.css");
</style>
<style type="text/css" media="all">
@import url("style.css%3Fou07bk.css");
</style>
<style type="text/css" media="print">
@import url("print.css%3Fou07bk.css");
</style>

<!--[if lt IE 7]>
<link type="text/css" rel="stylesheet" href="http://sbsweb2-bak.bio.ed.ac.uk/beast/themes/garland/fix-ie.css?ou07bk" media="all" />
<![endif]-->
  <script type="text/javascript" src="jquery.js%3Fv=1.4.4"></script>
<script type="text/javascript" src="jquery.once.js%3Fv=1.2"></script>
<script type="text/javascript" src="drupal.js%3Fou07bk"></script>
<script type="text/javascript">
<!--//--><![CDATA[//><!--
jQuery.extend(Drupal.settings, {"basePath":"\/beast\/","pathPrefix":"","ajaxPageState":{"theme":"garland","theme_token":"_PvE9Dn22qwdh_tm-C00uzu1pe0kMW8HYzgMLx9x6C8","js":{"misc\/jquery.js":1,"misc\/jquery.once.js":1,"misc\/drupal.js":1},"css":{"modules\/system\/system.base.css":1,"modules\/system\/system.menus.css":1,"modules\/system\/system.messages.css":1,"modules\/system\/system.theme.css":1,"modules\/field\/theme\/field.css":1,"modules\/node\/node.css":1,"modules\/search\/search.css":1,"modules\/user\/user.css":1,"sites\/all\/modules\/ckeditor\/css\/ckeditor.css":1,"themes\/garland\/style.css":1,"themes\/garland\/print.css":1,"themes\/garland\/fix-ie.css":1}}});
//--><!]]>
</script>
</head>
<body class="html not-front not-logged-in no-sidebars page-node page-node- page-node-52 node-type-article fluid-width" >
  <div id="skip-link">
    <a href="Discrete-Phylogeographic-Analysis.html#main-content" class="element-invisible element-focusable">Skip to main content</a>
  </div>
      
  <div id="wrapper">
    <div id="container" class="clearfix">

      <div id="header">
        <div id="logo-floater">
                              <div id="branding"><strong><a href="index.html">
                          <img src="beast.png" alt="BEAST " title="BEAST " id="logo" />
                        <span>BEAST</span>            </a></strong></div>
                          </div>

        <h2 class="element-invisible">Main menu</h2><ul class="links inline main-menu"><li class="menu-230 first"><a href="index.html">Home</a></li>
<li class="menu-307"><a href="programs.html">Programs</a></li>
<li class="menu-329"><a href="downloads.html">Downloads</a></li>
<li class="menu-312"><a href="tutorials.html" title="Tutorials and documentation on using BEAST">Tutorials</a></li>
<li class="menu-326"><a href="error-messages.html">Error Messages</a></li>
<li class="menu-313 last"><a href="faq.html" title="Frequently Asked Questions">FAQ</a></li>
</ul>              </div> <!-- /#header -->

      
      <div id="center"><div id="squeeze"><div class="right-corner"><div class="left-corner">
          <h2 class="element-invisible">You are here</h2><div class="breadcrumb"><a href="index.html">Home</a></div>                    <a id="main-content"></a>
          <div id="tabs-wrapper" class="clearfix">                                <h1 class="with-tabs">Discrete Phylogeographic Analysis</h1>
                              </div>                                                  <div class="clearfix">
            <div id="node-52" class="node node-article node-promoted" about="/beast/discrete-phylogeographic-analysis" typeof="sioc:Item foaf:Document">

  
      <span property="dc:title" content="Discrete Phylogeographic Analysis" class="rdf-meta element-hidden"></span>
      <span class="submitted"><span property="dc:date dc:created" content="2015-01-29T07:17:14+00:00" datatype="xsd:dateTime" rel="sioc:has_creator">Thu, 01/29/2015 - 07:17 — <span class="username" xml:lang="" about="/beast/users/rambaut" typeof="sioc:UserAccount" property="foaf:name" datatype="">rambaut</span></span></span>
  
  <div class="content clearfix">
    <div class="field field-name-body field-type-text-with-summary field-label-hidden"><div class="field-items"><div class="field-item even" property="content:encoded"><div id="bodyContent" style="color: rgb(0, 0, 0); font-family: sans-serif; line-height: 19.049999237060547px;">
<div id="contentSub" style="font-size: 11px; line-height: 1.2em; margin: 0px 0px 1.4em 1em; color: rgb(125, 125, 125); width: auto;"><a href="home.html" style="line-height: 19.049999237060547px; font-size: 13px; text-decoration: none; color: rgb(90, 54, 150); background-image: none;" title="Main Page">BEAST Documentation</a><span style="color:rgb(0, 0, 0); font-size:13px; line-height:19.049999237060547px">-&gt;</span><a href="Tutorials.html" style="line-height: 19.049999237060547px; font-size: 13px; text-decoration: none; color: rgb(90, 54, 150); background-image: none;" title="Tutorials">Tutorials</a><span style="color:rgb(0, 0, 0); font-size:13px; line-height:19.049999237060547px">-&gt; Discrete Phylogeographic Analysis</span></div>
<h2>Setting up a standard discrete phylogeographic analysis</h2>
<p>This tutorial describes how to edit a BEAST XML file to set up a discrete phylogeographic analysis using a standard continuous-time Markov chain (CTMC, <a class="external autonumber" href="https://web.archive.org/web/20120903075302/http://www.ploscompbiol.org/doi/pcbi.1000520" rel="nofollow" style="text-decoration: none; color: rgb(51, 102, 187); background-image: url(https://web.archive.org/web/20120722155546/http://beast.bio.ed.ac.uk/w/skins/monobook/external.png); padding: 0px 13px 0px 0px; background-position: 100% 50%; background-repeat: no-repeat no-repeat;" title="http://www.ploscompbiol.org/doi/pcbi.1000520">[1]</a>). The tutorial assumes you are using BEASTv1.2 or higher. The example XML file for this analysis can be found here <a class="external autonumber" href="https://web.archive.org/web/20120903075302/http://code.google.com/p/beast-mcmc/source/browse/trunk/examples/release/Phylogeography/H5N1_HA_discrete.xml" rel="nofollow" style="text-decoration: none; color: rgb(51, 102, 187); background-image: url(https://web.archive.org/web/20120722155546/http://beast.bio.ed.ac.uk/w/skins/monobook/external.png); padding: 0px 13px 0px 0px; background-position: 100% 50%; background-repeat: no-repeat no-repeat;" title="http://code.google.com/p/beast-mcmc/source/browse/trunk/examples/release/Phylogeography/H5N1_HA_discrete.xml">[2]</a> and considers influenza A H5N1 diffusion among (<em>K</em>=7) locations. This tutorial sets up a standard CTMC, but points out the necessary additions for Bayesian stochastic search variable selection (<a href="https://web.archive.org/web/20120903075302/http://beast.bio.ed.ac.uk/BSSVS" style="text-decoration: none; color: rgb(90, 54, 150); background-image: none; background-position: initial initial; background-repeat: initial initial;" title="BSSVS">BSSVS</a>, <a class="external autonumber" href="https://web.archive.org/web/20120903075302/http://www.ploscompbiol.org/doi/pcbi.1000520" rel="nofollow" style="text-decoration: none; color: rgb(51, 102, 187); background-image: url(https://web.archive.org/web/20120722155546/http://beast.bio.ed.ac.uk/w/skins/monobook/external.png); padding: 0px 13px 0px 0px; background-position: 100% 50%; background-repeat: no-repeat no-repeat;" title="http://www.ploscompbiol.org/doi/pcbi.1000520">[3]</a>) as well; a separate tutorial focuses on this procedure. Start by adding K location XML elements to the XML file (before the taxa is a good place):</p>
<pre>
	&lt;!-- The list of sampling locations                                        --&gt;
	&lt;!-- K = 7                                                                 --&gt;
	&lt;location id="Fujian"/&gt; 
	&lt;location id="Guangdong"/&gt; 
	&lt;location id="Guangxi"/&gt; 
	&lt;location id="Hebei"/&gt; 
	&lt;location id="Henan"/&gt; 
	&lt;location id="HongKong"/&gt; 
	&lt;location id="Hunan"/&gt; 
</pre><p>Add a location attribute to each taxon:</p>
<pre>
	&lt;taxa id="taxa"&gt;
		&lt;taxon id="A_chicken_Fujian_1042_2005"&gt;
			&lt;date value="2005.0" direction="forwards" units="years"/&gt;
			&lt;attr name="location"&gt;Fujian&lt;/attr&gt;
		&lt;/taxon&gt;
		&lt;taxon id="A_Duck_Fujian_1734_2005"&gt;
			&lt;date value="2005.0" direction="forwards" units="years"/&gt;
			&lt;attr name="location"&gt;Fujian&lt;/attr&gt;
		&lt;/taxon&gt;
		.
		.
		.
		&lt;taxon id="A_blackbird_Hunan_1_2004"&gt;
			&lt;date value="2004.0" direction="forwards" units="years"/&gt;
			&lt;attr name="location"&gt;Hunan&lt;/attr&gt;
		&lt;/taxon&gt;
	&lt;/taxa&gt;
</pre><p>Set up a general data type and a unique site pattern for the location data (after the alignment patterns would be good place):</p>
<pre>
	&lt;!-- The general data type for locations                                   --&gt;
	&lt;generalDataType id="geography"&gt;
		&lt;location idref="Fujian"/&gt; 
		&lt;location idref="Guangdong"/&gt; 
		&lt;location idref="Guangxi"/&gt; 
		&lt;location idref="Hebei"/&gt; 
		&lt;location idref="Henan"/&gt; 
		&lt;location idref="HongKong"/&gt; 
		&lt;location idref="Hunan"/&gt; 
	&lt;/generalDataType&gt;

	&lt;!-- 1 unique site pattern for the locations                                   --&gt;
	&lt;attributePatterns id="geoPatterns" attribute="location"&gt;
		&lt;generalDataType idref="geography"/&gt;
		&lt;taxa idref="taxa"/&gt;
	&lt;/attributePatterns&gt;
</pre><p>Add a svsGeneralSubstitutionModel and associated statistics (maybe after the treeLikelihood). The svsGeneralSubstitutionModel sets up the CTMC, equivalent to e.g. GTR for nucleotide substitution. The dimension of the frequencies in the CTMC needs to match the number of locations (<em>K</em>). Set the dimensions of the rates and indicators to <em>K</em>*(<em>K</em>-1)/2. The sumStatistic and productStatistic will be useful for the <a href="https://web.archive.org/web/20120903075302/http://beast.bio.ed.ac.uk/BSSVS" style="text-decoration: none; color: rgb(90, 54, 150); background-image: none; background-position: initial initial; background-repeat: initial initial;" title="BSSVS">BSSVS</a> extension of the standard model.</p>
<pre>
	&lt;!-- CTMC model for discrete state reconstructions             --&gt;
	&lt;svsGeneralSubstitutionModel id="originModel" name="origin"&gt;
		&lt;dataType idref="geography"/&gt;
		&lt;frequencies&gt;
			&lt;frequencyModel id="geoFreqs" normalize="true"&gt;
				&lt;dataType idref="geography"/&gt;
				&lt;frequencies&gt;
					&lt;parameter id="geoFreqs.frequencies" dimension="7"/&gt;
				&lt;/frequencies&gt;
			&lt;/frequencyModel&gt;
		&lt;/frequencies&gt;
		&lt;rates relativeTo="1"&gt;
			&lt;parameter id="rates" dimension="21" value="1.0"/&gt;
		&lt;/rates&gt;
		&lt;rateIndicator&gt;
			&lt;parameter id="indicators" dimension="21" value="1.0"/&gt;
		&lt;/rateIndicator&gt;
	&lt;/svsGeneralSubstitutionModel&gt;

	&lt;sumStatistic id="nonZeroRates" name="nonZeroRateCount" elementwise="true"&gt;
		&lt;parameter idref="indicators"/&gt;
	&lt;/sumStatistic&gt;

	&lt;productStatistic id="actualRates" elementwise="false"&gt;
		&lt;parameter idref="indicators"/&gt;
		&lt;parameter idref="rates"/&gt;
	&lt;/productStatistic&gt;
</pre><p>Similar to the likelihood approach of nucleotide substitution, complete the model specification by adding a siteModel and phylogenetic likelihood (ancestralTreeLikelihood).</p>
<pre>
    &lt;siteModel id="geoSiteModel"&gt;
        &lt;substitutionModel&gt;
            &lt;svsGeneralSubstitutionModel idref="originModel"/&gt;
        &lt;/substitutionModel&gt;
        &lt;mutationRate&gt;
            &lt;parameter id="geoSiteModel.mu" value="0.1" lower="0.0" upper="10.0"/&gt;
        &lt;/mutationRate&gt;
    &lt;/siteModel&gt;

    &lt;ancestralTreeLikelihood id="geoTreeLikelihood"&gt;
        &lt;patterns idref="geoPatterns"/&gt;
        &lt;treeModel idref="treeModel"/&gt;
        &lt;siteModel idref="geoSiteModel"/&gt;
        &lt;svsGeneralSubstitutionModel idref="originModel"/&gt;
    &lt;/ancestralTreeLikelihood&gt;
</pre><p>The additional parameters we need to sample are the geoSiteModel.mu and rates. Therefore, complete the operator list with two scaleOperator elements, one on geoSiteModel.mu and one on the rates; allow both to have a relatively high weight, in particular the operator on the rates. The two operators at the bottom are required for <a href="https://web.archive.org/web/20120903075302/http://beast.bio.ed.ac.uk/BSSVS" style="text-decoration: none; color: rgb(90, 54, 150); background-image: none; background-position: initial initial; background-repeat: initial initial;" title="BSSVS">BSSVS</a>, but commented out for the time being.</p>
<pre>
	&lt;operators id="operators" optimizationSchedule="log"&gt;
		.
		.
		.
		&lt;!-- discrete state model operators --&gt;
		&lt;scaleOperator scaleFactor="0.75" weight="10"&gt;
			&lt;parameter idref="geoSiteModel.mu"/&gt;
		&lt;/scaleOperator&gt;
		&lt;scaleOperator scaleFactor="0.75" weight="30" scaleAllIndependently="true" autoOptimize="true"&gt;
			&lt;parameter idref="rates"/&gt;
		&lt;/scaleOperator&gt;

		&lt;!-- BSSVS operators	--&gt;
		&lt;!--
		&lt;bitFlipOperator weight="30"&gt;
			&lt;parameter idref="indicators"/&gt;
		&lt;/bitFlipOperator&gt;
		&lt;bitFlipInSubstitutionModelOperator scaleFactor="0.75" weight="30" autoOptimize="true"&gt;
			&lt;svsGeneralSubstitutionModel idref="originModel"/&gt;
			&lt;parameter idref="geoSiteModel.mu"/&gt;
		&lt;/bitFlipInSubstitutionModelOperator&gt;             --&gt;
	&lt;/operators&gt;
</pre><p>For the same parameters, we need to specify priors. To this purpose, we propose a gammaPrior (with shape = scale = 1.0) for the rates and an exponentialPrior (mean = 1) for geoSiteModel.mu. The additional poissonPrior is commented out and only used for <a href="https://web.archive.org/web/20120903075302/http://beast.bio.ed.ac.uk/BSSVS" style="text-decoration: none; color: rgb(90, 54, 150); background-image: none; background-position: initial initial; background-repeat: initial initial;" title="BSSVS">BSSVS</a>. Also put the model reference in the prior: &lt;svsGeneralSubstitutionModel idref="originModel"/&gt;. </p>
<pre>
			&lt;prior id="prior"&gt;
				.
				.
				.
				&lt;cachedPrior&gt;
					&lt;gammaPrior shape="1.0" scale="1.0" offset="0.0"&gt;
						&lt;parameter idref="rates"/&gt;
					&lt;/gammaPrior&gt;
					&lt;parameter idref="rates"/&gt;
				&lt;/cachedPrior&gt;
				&lt;exponentialPrior mean="1" offset="0"&gt;
					&lt;parameter idref="geoSiteModel.mu"/&gt;
				&lt;/exponentialPrior&gt;
				&lt;!-- BSSVS truncated Poisson prior	--&gt;
				&lt;!--
				&lt;poissonPrior mean="0.693" offset="6.0"&gt;
					&lt;statistic idref="nonZeroRates"/&gt;
				&lt;/poissonPrior&gt;
				--&gt;
				&lt;svsGeneralSubstitutionModel idref="originModel"/&gt;
			&lt;/prior&gt;
</pre><p>The additional phylogenetic likelihood we specified above needs to be added to the likelihood element.</p>
<pre>
			&lt;likelihood id="likelihood"&gt;
				&lt;treeLikelihood idref="treeLikelihood"/&gt;
				&lt;ancestralTreeLikelihood idref="geoTreeLikelihood"/&gt;
			&lt;/likelihood&gt;
</pre><p>Because the rates and indicators vectors may contain many parameters, we log those in a separate log file together with the geoSiteModel.mu, which completes the CTMC rate matrix:</p>
<pre>
		 &lt;log id="rateMatrixLog" logEvery="10000" fileName="H5N1_HA_discrete_rateMatrix.log"&gt;
			&lt;parameter idref="rates"/&gt;
			&lt;parameter idref="indicators"/&gt;
   			&lt;sumStatistic idref="nonZeroRates"/&gt;
   			&lt;parameter idref="geoSiteModel.mu"/&gt;
		&lt;/log&gt;
</pre><p>Note that in the standard model all indicators will be fixed at their initial value (=1), but logging those will be useful for <a href="https://web.archive.org/web/20120903075302/http://beast.bio.ed.ac.uk/BSSVS" style="text-decoration: none; color: rgb(90, 54, 150); background-image: none; background-position: initial initial; background-repeat: initial initial;" title="BSSVS">BSSVS</a>. For diagnostic purposes, we can also add ancestralTreeLikelihood and geoSiteModel.mu to the regular parameter log file.</p>
<pre>
		&lt;log id="fileLog" logEvery="10000" fileName="H5N1_HA_discrete.log"&gt;
			.
			.
			.
			&lt;parameter idref="geoSiteModel.mu"/&gt;
			&lt;ancestralTreeLikelihood idref="geoTreeLikelihood"/&gt;
		&lt;/log&gt;
</pre><p>Finally, to annotate the trees with with the discrete reconstructions using the tree-traversal alorithm we need to add ancestralTreeLikelihood to the logTree element:</p>
<pre>
		&lt;logTree id="treeFileLog" logEvery="10000" nexusFormat="true" fileName="H5N1_HA_discrete.trees" sortTranslationTable="true"&gt;
			&lt;treeModel idref="treeModel"/&gt;
			&lt;discretizedBranchRates idref="branchRates"/&gt;
			&lt;!-- this entry annotates trees with the discrete reconstructions --&gt;
			&lt;ancestralTreeLikelihood idref="geoTreeLikelihood"/&gt;
			&lt;posterior idref="posterior"/&gt;
		&lt;/logTree&gt;
</pre><hr /><p><a href="Tutorials.html" style="text-decoration: none; color: rgb(90, 54, 150); background-image: none; background-position: initial initial; background-repeat: initial initial;" title="Tutorials">Return to Tutorials</a></p>
<p>Philippe Lemey, 2009</p>
</div>
</div></div></div>  </div>

  <div class="clearfix">
          <div class="links"></div>
    
      </div>

</div>
          </div>
                          </div></div></div></div> <!-- /.left-corner, /.right-corner, /#squeeze, /#center -->

      
    </div> <!-- /#container -->
  </div> <!-- /#wrapper -->
  </body>
</html>
