<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML+RDFa 1.0//EN"
  "http://www.w3.org/MarkUp/DTD/xhtml-rdfa-1.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" version="XHTML+RDFa 1.0" dir="ltr"
  xmlns:content="http://purl.org/rss/1.0/modules/content/"
  xmlns:dc="http://purl.org/dc/terms/"
  xmlns:foaf="http://xmlns.com/foaf/0.1/"
  xmlns:og="http://ogp.me/ns#"
  xmlns:rdfs="http://www.w3.org/2000/01/rdf-schema#"
  xmlns:sioc="http://rdfs.org/sioc/ns#"
  xmlns:sioct="http://rdfs.org/sioc/types#"
  xmlns:skos="http://www.w3.org/2004/02/skos/core#"
  xmlns:xsd="http://www.w3.org/2001/XMLSchema#">

<head profile="http://www.w3.org/1999/xhtml/vocab">
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<link rel="shortcut icon" href="BEAST_favicon.ico" type="image/vnd.microsoft.icon" />
<link rel="shortlink" href="10.html" />
<link rel="canonical" href="tutorial-1.html" />
<meta name="Generator" content="Drupal 7 (http://drupal.org)" />
  <title>Tutorial 1 | BEAST</title>
  <style type="text/css" media="all">
@import url("system.base.css%3Fou07bk.css");
@import url("system.menus.css%3Fou07bk.css");
@import url("system.messages.css%3Fou07bk.css");
@import url("system.theme.css%3Fou07bk.css");
</style>
<style type="text/css" media="all">
@import url("field.css%3Fou07bk.css");
@import url("node.css%3Fou07bk.css");
@import url("search.css%3Fou07bk.css");
@import url("user.css%3Fou07bk.css");
@import url("ckeditor.css%3Fou07bk.css");
</style>
<style type="text/css" media="all">
@import url("style.css%3Fou07bk.css");
</style>
<style type="text/css" media="print">
@import url("print.css%3Fou07bk.css");
</style>

<!--[if lt IE 7]>
<link type="text/css" rel="stylesheet" href="http://sbsweb2-bak.bio.ed.ac.uk/beast/themes/garland/fix-ie.css?ou07bk" media="all" />
<![endif]-->
  <script type="text/javascript" src="jquery.js%3Fv=1.4.4"></script>
<script type="text/javascript" src="jquery.once.js%3Fv=1.2"></script>
<script type="text/javascript" src="drupal.js%3Fou07bk"></script>
<script type="text/javascript">
<!--//--><![CDATA[//><!--
jQuery.extend(Drupal.settings, {"basePath":"\/beast\/","pathPrefix":"","ajaxPageState":{"theme":"garland","theme_token":"M-R-7-ndEHoTcs5bqMQZXRggO2-iTb0v2ZHGm9U_uG0","js":{"misc\/jquery.js":1,"misc\/jquery.once.js":1,"misc\/drupal.js":1},"css":{"modules\/system\/system.base.css":1,"modules\/system\/system.menus.css":1,"modules\/system\/system.messages.css":1,"modules\/system\/system.theme.css":1,"modules\/field\/theme\/field.css":1,"modules\/node\/node.css":1,"modules\/search\/search.css":1,"modules\/user\/user.css":1,"sites\/all\/modules\/ckeditor\/css\/ckeditor.css":1,"themes\/garland\/style.css":1,"themes\/garland\/print.css":1,"themes\/garland\/fix-ie.css":1}}});
//--><!]]>
</script>
</head>
<body class="html not-front not-logged-in no-sidebars page-node page-node- page-node-10 node-type-article fluid-width" >
  <div id="skip-link">
    <a href="tutorial-1.html#main-content" class="element-invisible element-focusable">Skip to main content</a>
  </div>
      
  <div id="wrapper">
    <div id="container" class="clearfix">

      <div id="header">
        <div id="logo-floater">
                              <div id="branding"><strong><a href="index.html">
                          <img src="beast.png" alt="BEAST " title="BEAST " id="logo" />
                        <span>BEAST</span>            </a></strong></div>
                          </div>

        <h2 class="element-invisible">Main menu</h2><ul class="links inline main-menu"><li class="menu-230 first"><a href="index.html">Home</a></li>
<li class="menu-307"><a href="programs.html">Programs</a></li>
<li class="menu-329"><a href="downloads.html">Downloads</a></li>
<li class="menu-312 active-trail"><a href="tutorials.html" title="Tutorials and documentation on using BEAST" class="active-trail">Tutorials</a></li>
<li class="menu-326"><a href="error-messages.html">Error Messages</a></li>
<li class="menu-313 last"><a href="faq.html" title="Frequently Asked Questions">FAQ</a></li>
</ul>              </div> <!-- /#header -->

      
      <div id="center"><div id="squeeze"><div class="right-corner"><div class="left-corner">
          <h2 class="element-invisible">You are here</h2><div class="breadcrumb"><a href="index.html">Home</a> › <a href="tutorials.html" title="Tutorials and documentation on using BEAST">Tutorials</a></div>                    <a id="main-content"></a>
          <div id="tabs-wrapper" class="clearfix">                                <h1 class="with-tabs">Tutorial 1</h1>
                              </div>                                                  <div class="clearfix">
            <div id="node-10" class="node node-article node-promoted" about="/beast/tutorial-1" typeof="sioc:Item foaf:Document">

  
      <span property="dc:title" content="Tutorial 1" class="rdf-meta element-hidden"></span>
      <span class="submitted"><span property="dc:date dc:created" content="2014-08-14T10:04:42+01:00" datatype="xsd:dateTime" rel="sioc:has_creator">Thu, 08/14/2014 - 10:04 — <span class="username" xml:lang="" about="/beast/users/rambaut" typeof="sioc:UserAccount" property="foaf:name" datatype="">rambaut</span></span></span>
  
  <div class="content clearfix">
    <div class="field field-name-body field-type-text-with-summary field-label-hidden"><div class="field-items"><div class="field-item even" property="content:encoded"><h1>Using sequences sampled at different points in time to estimate rates</h1>
<p>This document provides a step-by-step tutorial to converting a NEXUS file with a DATA or CHARACTERS block into a BEAST XML file. This is done using the program <a href="beauti.html">BEAUti</a> (this stands for Bayesian Evolutionary Analysis Utility). This is a user-friendly program for setting the evolutionary model and options for the MCMC.</p>
<p><a id="Running_BEAUti" name="Running_BEAUti"></a></p>
<h1>Running BEAUti</h1>
<p>The exact instructions for running BEAUti differs depending on which computer you are using. Please see the README text file that was distributed with the version you downloaded.</p>
<p>Once running, BEAUti will look similar irrespective of which computer system it is running on. For this tutorial, the Mac OS X version will be shown but the Linux &amp; Windows versions will have exactly the same layout and functionality.</p>
<p><a id="Loading_the_NEXUS_file" name="Loading_the_NEXUS_file"></a></p>
<h1>Loading the NEXUS file</h1>
<p>To load a NEXUS format alignment, simply select the Import NEXUS... option from the File menu:</p>
<p><a class="image" href="https://web.archive.org/web/20121219123805/http://beast.bio.ed.ac.uk/File:Beauti0.png" title="beauti0.png"><img alt="beauti0.png" src="https://web.archive.org/web/20121219123805im_/http://beast.bio.ed.ac.uk/w/images/1/1c/Beauti0.png" /></a></p>
<p><a id="The_NEXUS_alignment" name="The_NEXUS_alignment"></a></p>
<h2>The NEXUS alignment</h2>
<p>The examples folder contains a file called Dengue4.env.nex. This file contains an alignment of 17 sequences from the env gene of Dengue-4 virus from various parts of the world with dates from 1956-1994. It starts out like this (the lines have been truncated):</p>
<pre>
#NEXUS

BEGIN DATA;
DIMENSIONS NTAX=17 NCHAR=1485;
FORMAT MISSING=? GAP=- DATATYPE=DNA;
MATRIX
D4Brazi82       ATGCGATGCGTAGGAGTAGGAAACAGAGACTTTGTGGAAGGAGTCTCAGGTGGAGCATGGGTC...
D4ElSal83       ATGCGATGCGTAGGAGTAGGAAACAGAGACTTTGTGGAAGGAGTCTCAGGTGGAGCATGGGTC...
D4ElSal94       ATGCGATGCGTAGGAGTAGGAAACAGAGACTTTGTGGAAGGAGTCTCAGGTGGAGCATGGGTC...
D4Indon76       ATGCGATGCGTAGGAGTAGGAAACAGAGACTTTGTGGAAGGAGTCTCAGGTGGAGCATGGGTC...
D4Indon77       ATGCGATGCGTAGGAGTAGGAAACAGAGACTTTGTGGAAGGAGTCTCAGGTGGAGCATGGGTC...
D4Mexico84      ATGCGATGCGTAGGAGTAGGAAACAGAGACTTTGTGGAAGGAGTCTCAGGTGGAGCATGGGTC...
D4NewCal81      ATGCGATGCGTAGGAGTAGGAAACAGAGACTTTGTGGAAGGAGTCTCAGGTGGAGCATGGGTC...
D4Philip64      ATGCGATGCGTGGGAGTGGGGAACAGAGACTTTGTGGAAGGAGTCTCAGGTGGAGCATGGGTC...
D4Philip56      ATGCGATGCGTGGGAGTGGGGAACAGAGACTTTGTGGAAGGAGTCTCAGGTGGAGCATGGGTC...
D4Philip84      ATGCGATGCGTAGGAGTGGGGAACAGAGACTTTGTGGAAGGAGTCTCAGGTGGAGCATGGGTC...
D4PRico86       ATGCGATGCGTAGGAGTAGGAAACAGAGACTTTGTGGAAGGAGTCTCAGGTGGAGCATGGGTC...
D4SLanka78      ATGCGATGCGTGGGAGTGGGGAACAGAGACTTTGTGGAAGGAGTCTCAGGTGGAGCATGGGTC...
D4Tahiti79      ATGCGATGCGTAGGAGTAGGAAACAGAGACTTTGTGGAAGGAGTCTCAGGTGGAGCATGGGTC...
D4Tahiti85      ATGCGATGCGTAGGAGTAGGAAACAGAGACTTTGTGGAAGGAGTTTCAGGTGGAGCATGGGTC...
D4Thai63        ATGCGATGCGTAGGAGTGGGGAACAGGGACTTTGTGGAAGGAGTCTCAGGTGGAGCATGGGTC...
D4Thai78        ATGCGATGCGTAGGAGTGGGGAACAGAGACTTTGTAGAAGGAGTCTCAGGTGGAGCATGGGTC...
D4Thai84        ATGCGATGCGTAGGAGTAGGGAACAGAGACTTTGTAGAAGGAGTCTCAGGTGGAGCATGGGTC...
;
END;</pre><p>Once loaded, the list of taxa and the actual alignment will be displayed in the main window:</p>
<p><a class="image" href="https://web.archive.org/web/20121219123805/http://beast.bio.ed.ac.uk/File:Beauti1.png" title=" beauti1.png"><img alt=" beauti1.png" src="https://web.archive.org/web/20121219123805im_/http://beast.bio.ed.ac.uk/w/images/1/13/Beauti1.png" /></a></p>
<p><a id="Setting_the_dates_of_the_taxa" name="Setting_the_dates_of_the_taxa"></a></p>
<h2>Setting the dates of the taxa</h2>
<p>By default all the taxa are assumed to have a date of zero. This will be correct if all the sequences were sampled at approximately the same point in time (at least on the time scale of their evolutionary history). In this case, the Dengue-4 virus have been sampled at various dates going back to the 1950s. The actual year of sampling is given in the name of each taxon and we could simply edit the value in the Date column of the table to reflect these. However it is more convenient to use the Guess Dates button at the top of the window. Clicking this will make the following dialog box appear:</p>
<p><a class="image" href="https://web.archive.org/web/20121219123805/http://beast.bio.ed.ac.uk/File:Beauti2.png" title=" beauti2.png"><img alt=" beauti2.png" src="https://web.archive.org/web/20121219123805im_/http://beast.bio.ed.ac.uk/w/images/c/c3/Beauti2.png" /></a></p>
<p>This operation attempts to guess what the dates are from information contained within the taxon names. Generally this works by trying to find a numerical field within each name. If the taxon names contain more than one numerical field (such as the Dengue-4 sequences, above) then you can specify how to find the one that corresponds to the date of sampling. You can either specify the order that the date field comes (e.g., first, last or various positions in between) or specify a prefix (some characters that come immediately before the date field in each name). For the Dengue-4 sequences you select last from the drop-down menu for the order. If you left it at first then all taxa would be given a date of "4" taken from the second character of each name.</p>
<p>In this dialog box, you can also get BEAUti to add a fixed value to each guessed date. In this case the value "1900" has been added to turn the dates from 2 digit years to 4 digit. Any dates in the taxon names given as "00" would thus become "1900". If these dates were actually referring to the year 2000 then selecting the next option would convert them correctly, adding 2000 to any date less than 07.</p>
<p>When you press OK the dates will appear in the appropriate column of the main window. You can then check these and edit them manually as required. At the top of the window you can set the units that the dates are given in (years, months, days) and whether they are specified relative to a point in the past (as would be the case for years such as 1984) or backwards in time from the present.</p>
<p><a id="Translating_the_data_into_amino_acid_sequences" name="Translating_the_data_into_amino_acid_sequences"></a></p>
<h2>Translating the data into amino acid sequences</h2>
<p>At the bottom of the main window is the option to translate the data into amino acid sequences. This will be done using the genetic code specified in the associated drop down menu. If the loaded sequence are not nucleotides then this option will be disabled.</p>
<p><a id="Setting_the_evolutionary_model" name="Setting_the_evolutionary_model"></a></p>
<h1>Setting the evolutionary model</h1>
<p>The next thing to do is to click on the Model tab at the top of the main window. This will reveal the evolutionary model settings for BEAST. Exactly which options appear depend on whether the data are nucleotides or amino acids (or nucleotides translated into amino acids). The settings that will appear after loading the Dengue-4 data set will be as follows:</p>
<p><a class="image" href="https://web.archive.org/web/20121219123805/http://beast.bio.ed.ac.uk/File:Beauti3.png" title=" beauti3.png"><img alt=" beauti3.png" src="https://web.archive.org/web/20121219123805im_/http://beast.bio.ed.ac.uk/w/images/6/60/Beauti3.png" /></a></p>
<p>Substitution Model</p>
<p>Select between the HKY and the GTR model for nucleotide sequences or a variety of common amino acid substitution models.</p>
<p>Site Heterogeneity Model</p>
<p>Select between gamma or invariant sites or both.</p>
<p>Number of Gamma Categories</p>
<p>The number of categories for the descrete approximation to the gamma distribution.</p>
<p>Partition into codon positions</p>
<p>For when the data are nucleotide sequences that have been aligned to the reading frame. This allows a separate substitution rate for each codon position.</p>
<p>If selected then the following options can also be used:</p>
<p>Unlink substition model across codon positions</p>
<p>Allows a different substitution model for each codon position. This means a separate kappa (for HKY) or rate matrix (for GTR).</p>
<p>Unlink rate heterogeneity model across codon positions</p>
<p>Allows a different rate heterogeneity model for each codon position. This means a separate alpha (for gamma) or Pinv (for invariant sites).</p>
<p>Fix mean substitution rate</p>
<p>When dates have been associated with the sequences, the substitution rate can be estimated on an absolute time scale. If this is on then the substitution rate is fixed to the value in the text field below otherwise it is estimated from the data.</p>
<p>Mean Substitution Rate</p>
<p>If there are no dates associated with the sequences than this can be used to specify an _a_priori_ substition rate that will be used to scale all the times in the tree to units of time. This can be used to calibrate the tree with an external rate. Doing this will mean abandoning any errors associated with this rate.</p>
<p>Coalescent Model</p>
<p>Choose between a constant population size, an exponentially growing population or a logistically growing population.</p>
<p>This tutorial assumes you are familiar with the evolutionary models available, however there are a couple of points to note:</p>
<ol><li>Selecting the Partition into codon positions option assumes that the data are aligned as codons. This option will then estimate a separate rate of substitution for each codon position.</li>
<li>Selecting the Unlink substitution model across codon positions will specify that BEAST should estimate a separate transition-transversion ratio or general time reversible rate matrix for each codon position.</li>
<li>Selecting the Unlink rate heterogeneity model across codon positions will specify that BEAST should estimate set of rate heterogeneity parameters (gamma shape parameter and or proportion of invariant sites) for each codon position.</li>
<li>If you have specified dates of sampling for the sequences then you should select the Use dates associated with sequences option (this will be set by default if any taxa have dates that differ). You can then specify an initial estimate for the substitution rate. Setting this to be on the low side may help the MCMC reach equilibrium faster. If there are no dates for the sequences (they are contemporaneous) then you can specify a fixed over-all substitution rate obtained from another source. Setting this to 1.0 will result in the ages of the nodes of the tree being estimated in units of _substitutions per site_.</li>
</ol><p><a id="Setting_the_priors" name="Setting_the_priors"></a></p>
<h1>Setting the priors</h1>
<p>The next tab allows priors to be specified for each parameter in the model:</p>
<p><a class="image" href="https://web.archive.org/web/20121219123805/http://beast.bio.ed.ac.uk/File:Beauti4.png" title=" beauti4.png"><img alt=" beauti4.png" src="https://web.archive.org/web/20121219123805im_/http://beast.bio.ed.ac.uk/w/images/4/4c/Beauti4.png" /></a></p>
<p>In this tutorial, the default options will be used.</p>
<p><a id="Setting_up_the_operators" name="Setting_up_the_operators"></a></p>
<h1>Setting up the operators</h1>
<p>The next stage is to look at the operators for the MCMC. To do this select the Operators tab at the top of the main window. For the Dengue-4 dataset, with the model set up as shown in the screen shot above, you will see the following table:</p>
<p><a class="image" href="https://web.archive.org/web/20121219123805/http://beast.bio.ed.ac.uk/File:Beauti5.png" title=" beauti5.png"><img alt=" beauti5.png" src="https://web.archive.org/web/20121219123805im_/http://beast.bio.ed.ac.uk/w/images/4/42/Beauti5.png" /></a></p>
<p>Each parameter in the model has one or more "operators". The operators specify how the parameter changes as the MCMC runs. This table lists the parameters, their operators and the tuning settings for these operators. In the first column are the parameter names. These will be called things like hky1.kappa which means the <em>HKY</em> model's <em>kappa</em> parameter (the transition-transversion bias). The next column has the type of operators that are acting on each parameter. For example, the <em>scale operator</em> scales the parameter up or down by a proportion, the <em>random walk operator</em> adds or subtracts an amount to the parameter and the <em>uniform operator</em> simply picks a new value uniformally within a range. Some parameters relate to the tree or to the heights of the nodes of the tree and these have special operators.</p>
<p>The next column, labelled Tuning, gives a tuning setting to the operator. Some operators don't have any tuning settings so have n/a under this column. Changing the tuning setting will set how large a move that operator will make which will affect how often that change is accepted by the MCMC which will affect the efficency of the analysis. At the top of the window is an option called Auto Optimize which, when selected, will automatically adjust the tuning setting as the MCMC runs to try to achieve maximum efficiency. At the end of the run a table of the operators, their performance and the final values of these tuning settings will be displayed. These can then be used to set the starting tuning settings in order to minimize the amount of time taken to reach optimum performance in subsequent runs.</p>
<p>The next column, labelled Weight, specifies how often each operator is applied relative to each other. Some parameters have very little interaction with the rest of the model and as a result tend to be estimated very efficiently - an example is the <em>kappa</em>parameter - these parameters can have their operators down-weighted so that they are not changed as often.</p>
<p><a id="Setting_the_MCMC_options" name="Setting_the_MCMC_options"></a></p>
<h1>Setting the MCMC options</h1>
<p>The next tab, MCMC, provides settings to control the MCMC:</p>
<p><a class="image" href="https://web.archive.org/web/20121219123805/http://beast.bio.ed.ac.uk/File:Beauti6.png" title=" beauti6.png"><img alt=" beauti6.png" src="https://web.archive.org/web/20121219123805im_/http://beast.bio.ed.ac.uk/w/images/6/62/Beauti6.png" /></a></p>
<p>Firstly we have the Length of chain. This is the number of steps the MCMC will make in the chain before finishing. How long this should be depends on the size of the dataset, the complexity of the model and the quality of answer required. The default value of 10,000,000 is entirely arbitrary and should be adjusted according to the size of your dataset. In order examine whether a particular chain length is adequate, the resulting log file can be analysed using <a href="https://web.archive.org/web/20121219123805/http://beast.bio.ed.ac.uk/Tracer" title="Tracer">Tracer</a>. The aim of setting the chain length is to achieve a reasonable Effective Sample Size (ESS). Ways of doing this are discussed in [another tutorial|Increasing ESSs].</p>
<p>The next options specify how often the current parameter values should be displayed on the screen and recorded in the log file. The screen output is simply for monitoring the programs progress so can be set to any value (although if set too small, the sheer quantity of information being displayed on the screen may actually slow the program down). For the log file, the value should be set relative to the total length of the chain. Sampling too often will result in very large files with little extra benefit in terms of the precision of the analysis. Sample too infrequently and the log file will not contain much information about the distributions of the parameters. You probably want to aim to store about 10,000 samples so this should be set to the chain length / 10,000.</p>
<p>For this dataset let's initially setting the chain length to 100,000 as this will run reasonably quickly on most modern computers.</p>
<p>The final two options give the file names of the log files for the parameters and the trees. These will be set to a default based on the name of the imported NEXUS file but feel free to change these.</p>
<p><a id="Saving_and_Loading_BEAUti_files" name="Saving_and_Loading_BEAUti_files"></a></p>
<h1>Saving and Loading BEAUti files</h1>
<p>If you select the Save option from the File menu this will save a document in BEAUti's own format. Note that although this is an XML document it is not in the format that BEAST understands (it is much simpler and specific to the user interface of BEAUti). The idea is that the settings and data in BEAUti can be saved and loaded at a later time. We suggest you save BEAUti files with the extension '.beauti'.</p>
<p><a id="Generating_the_BEAST_XML_file" name="Generating_the_BEAST_XML_file"></a></p>
<h1>Generating the BEAST XML file</h1>
<p>We are now ready to create the BEAST XML file. Select Generate BEAST File... from the File menu (or click the button at the bottom of the window) and save the file with an appropriate name (we usually end the filename with '.xml'). We are now ready to run the file through BEAST.</p>
<p><a id="Running_BEAST" name="Running_BEAST"></a></p>
<h1>Running BEAST</h1>
<p>Now run BEAST according to the instructions provided (look at the README text file included in the package you downloaded), providing your newly created XML file as input.</p>
<pre>
+-----------------------------------------------\
|             BEAST v1.3 2002-2006              ||
| Bayesian Evolutionary Analysis Sampling Trees ||
|     BEAST Library: Build 1.10 2003/09/03      ||
|       Alexei Drummond and Andrew Rambaut      ||
|              University of Oxford             ||
|      http://evolve.zoo.ox.ac.uk/Beast/        ||
\-----------------------------------------------\|
 \-----------------------------------------------\
 +-----------------------------------------------+
 | Components created by:                        |
 |       Alexei Drummond                         |
 |       Roald Forsberg                          |
 |       Oliver Pybus                            |
 |       Andrew Rambaut                          |
 | Thanks to (for use of their code):            |
 |       Korbinian Strimmer                      |
 |       Oliver Pybus                            |
 +-----------------------------------------------+

Read alignment:
  Sequences = 17
      Sites = 1485
   Datatype = nucleotide
Site patterns 'patterns1' created from positions 1-1485 of alignment 'alignment'
  only using every 3 site
  pattern count = 34
Site patterns 'patterns2' created from positions 2-1485 of alignment 'alignment'
  only using every 3 site
  pattern count = 17
Site patterns 'patterns3' created from positions 3-1485 of alignment 'alignment'
  only using every 3 site
  pattern count = 119
Native nucleotide likelihood core found
TreeLikelihood using native nucleotide likelihood core.
TreeLikelihood using native nucleotide likelihood core.
TreeLikelihood using native nucleotide likelihood core.
Pre-burnin...
state	likelihood	treeModel.rootHeight	treeLikelihood1	treeLikelihood2	treeLikelihood3	coalescent
0	-3,919.1283	159.9998		-995.1892	-838.3758	-1,989.881	-95.6820
1000	-3,806.6080	125.4206		-971.7499	-828.9929	-1,917.5387	-88.3264
2000	-3,784.8661	96.7894			-966.8750	-829.8393	-1,904.5461	-83.6057
.
.
.
</pre><pre>
Operator analysis
Operator                                      Pr(accept)  Performance suggestion
hky.kappa                             0.331   0.2007      good
siteModel1.mu                         0.442   0.2753      good
siteModel2.mu                         0.319   0.2594      good
siteModel3.mu                         0.658   0.2326      good
up:compoundParameter down:nodeHeights 0.885   0.1548      good
constant.popSize                      0.293   0.2561      good
treeModel.rootHeight                  0.697   0.1897      slightly low	Try setting scaleFactor to about 0.7461
nodeHeights                                   0.4338      high
subtreeSlide                          7.198   0.1951      slightly low	Try decreasing size to about 6.001352499826204
Narrow Exchange                               0.2655      good
Wide Exchange                                 0.0543      good
wilsonBalding                                 0.0139      slightly low

23.212 seconds
</pre><hr /><p><a href="Tutorials.html" title="Tutorials">Return to Tutorials</a></p>
</div></div></div>  </div>

  <div class="clearfix">
          <div class="links"></div>
    
      </div>

</div>
          </div>
                          </div></div></div></div> <!-- /.left-corner, /.right-corner, /#squeeze, /#center -->

      
    </div> <!-- /#container -->
  </div> <!-- /#wrapper -->
  </body>
</html>
