<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML+RDFa 1.0//EN"
  "http://www.w3.org/MarkUp/DTD/xhtml-rdfa-1.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" version="XHTML+RDFa 1.0" dir="ltr"
  xmlns:content="http://purl.org/rss/1.0/modules/content/"
  xmlns:dc="http://purl.org/dc/terms/"
  xmlns:foaf="http://xmlns.com/foaf/0.1/"
  xmlns:og="http://ogp.me/ns#"
  xmlns:rdfs="http://www.w3.org/2000/01/rdf-schema#"
  xmlns:sioc="http://rdfs.org/sioc/ns#"
  xmlns:sioct="http://rdfs.org/sioc/types#"
  xmlns:skos="http://www.w3.org/2004/02/skos/core#"
  xmlns:xsd="http://www.w3.org/2001/XMLSchema#">

<head profile="http://www.w3.org/1999/xhtml/vocab">
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<link rel="shortcut icon" href="BEAST_favicon.ico" type="image/vnd.microsoft.icon" />
<link rel="shortlink" href="15.html" />
<link rel="canonical" href="tutorial-2.html" />
<meta name="Generator" content="Drupal 7 (http://drupal.org)" />
  <title>Tutorial 2 | BEAST</title>
  <style type="text/css" media="all">
@import url("system.base.css%3Fou07bk.css");
@import url("system.menus.css%3Fou07bk.css");
@import url("system.messages.css%3Fou07bk.css");
@import url("system.theme.css%3Fou07bk.css");
</style>
<style type="text/css" media="all">
@import url("field.css%3Fou07bk.css");
@import url("node.css%3Fou07bk.css");
@import url("search.css%3Fou07bk.css");
@import url("user.css%3Fou07bk.css");
@import url("ckeditor.css%3Fou07bk.css");
</style>
<style type="text/css" media="all">
@import url("style.css%3Fou07bk.css");
</style>
<style type="text/css" media="print">
@import url("print.css%3Fou07bk.css");
</style>

<!--[if lt IE 7]>
<link type="text/css" rel="stylesheet" href="http://sbsweb2-bak.bio.ed.ac.uk/beast/themes/garland/fix-ie.css?ou07bk" media="all" />
<![endif]-->
  <script type="text/javascript" src="jquery.js%3Fv=1.4.4"></script>
<script type="text/javascript" src="jquery.once.js%3Fv=1.2"></script>
<script type="text/javascript" src="drupal.js%3Fou07bk"></script>
<script type="text/javascript">
<!--//--><![CDATA[//><!--
jQuery.extend(Drupal.settings, {"basePath":"\/beast\/","pathPrefix":"","ajaxPageState":{"theme":"garland","theme_token":"ngxHmTTScM3Y3bHY2s9m_7HNtjXDb0b3I7QJQRpA7ak","js":{"misc\/jquery.js":1,"misc\/jquery.once.js":1,"misc\/drupal.js":1},"css":{"modules\/system\/system.base.css":1,"modules\/system\/system.menus.css":1,"modules\/system\/system.messages.css":1,"modules\/system\/system.theme.css":1,"modules\/field\/theme\/field.css":1,"modules\/node\/node.css":1,"modules\/search\/search.css":1,"modules\/user\/user.css":1,"sites\/all\/modules\/ckeditor\/css\/ckeditor.css":1,"themes\/garland\/style.css":1,"themes\/garland\/print.css":1,"themes\/garland\/fix-ie.css":1}}});
//--><!]]>
</script>
</head>
<body class="html not-front not-logged-in no-sidebars page-node page-node- page-node-15 node-type-article fluid-width" >
  <div id="skip-link">
    <a href="tutorial-2.html#main-content" class="element-invisible element-focusable">Skip to main content</a>
  </div>
      
  <div id="wrapper">
    <div id="container" class="clearfix">

      <div id="header">
        <div id="logo-floater">
                              <div id="branding"><strong><a href="index.html">
                          <img src="beast.png" alt="BEAST " title="BEAST " id="logo" />
                        <span>BEAST</span>            </a></strong></div>
                          </div>

        <h2 class="element-invisible">Main menu</h2><ul class="links inline main-menu"><li class="menu-230 first"><a href="index.html">Home</a></li>
<li class="menu-307"><a href="programs.html">Programs</a></li>
<li class="menu-329"><a href="downloads.html">Downloads</a></li>
<li class="menu-312 active-trail"><a href="tutorials.html" title="Tutorials and documentation on using BEAST" class="active-trail">Tutorials</a></li>
<li class="menu-326"><a href="error-messages.html">Error Messages</a></li>
<li class="menu-313 last"><a href="faq.html" title="Frequently Asked Questions">FAQ</a></li>
</ul>              </div> <!-- /#header -->

      
      <div id="center"><div id="squeeze"><div class="right-corner"><div class="left-corner">
          <h2 class="element-invisible">You are here</h2><div class="breadcrumb"><a href="index.html">Home</a> › <a href="tutorials.html" title="Tutorials and documentation on using BEAST">Tutorials</a></div>                    <a id="main-content"></a>
          <div id="tabs-wrapper" class="clearfix">                                <h1 class="with-tabs">Tutorial 2</h1>
                              </div>                                                  <div class="clearfix">
            <div id="node-15" class="node node-article node-promoted" about="/beast/tutorial-2" typeof="sioc:Item foaf:Document">

  
      <span property="dc:title" content="Tutorial 2" class="rdf-meta element-hidden"></span>
      <span class="submitted"><span property="dc:date dc:created" content="2014-08-14T11:38:02+01:00" datatype="xsd:dateTime" rel="sioc:has_creator">Thu, 08/14/2014 - 11:38 — <span class="username" xml:lang="" about="/beast/users/rambaut" typeof="sioc:UserAccount" property="foaf:name" datatype="">rambaut</span></span></span>
  
  <div class="content clearfix">
    <div class="field field-name-body field-type-text-with-summary field-label-hidden"><div class="field-items"><div class="field-item even" property="content:encoded"><p><a href="home.html" title="Main Page">BEAST Documentation</a>-&gt;<a href="tutorials.html" title="Tutorials">Tutorials</a>-&gt; Tutorial 2</p>
<h1>Estimating the divergence time of a monophyletic group using a known mutation rate</h1>
<p>This tutorial describes the use of BEAUti and BEAST to analyse some primate sequences and estimate the date of divergence of humans and chimps when the mutation rate is known.</p>
<p><a id="Running_BEAUti" name="Running_BEAUti"></a></p>
<h2>Running BEAUti</h2>
<p>The exact instructions for running BEAUti differs depending on which computer you are using. Please see the README text file that was distributed with the version you downloaded.</p>
<p>Once running, BEAUti will look similar irrespective of which computer system it is running on. For this tutorial, the Mac OS X version will be shown but the Linux &amp; Windows versions will have exactly the same layout and functionality.</p>
<p><a id="Loading_the_NEXUS_file" name="Loading_the_NEXUS_file"></a></p>
<h2>Loading the NEXUS file</h2>
<p>To load a NEXUS format alignment, simply select the Import NEXUS... option from the File menu:</p>
<p><a class="image" href="https://web.archive.org/web/20121204042736/http://beast.bio.ed.ac.uk/File:Tutorial2Beauti0.png" title="tutorial2Beauti0.png"><img alt="tutorial2Beauti0.png" src="https://web.archive.org/web/20121204042736im_/http://beast.bio.ed.ac.uk/w/images/7/7b/Tutorial2Beauti0.png" /></a></p>
<p><a id="The_NEXUS_alignment" name="The_NEXUS_alignment"></a></p>
<h3>The NEXUS alignment</h3>
<p>The examples folder contains a file called Primates.nex. This file contains an alignment of mitochondrial tRNA sequences from 6 primate species. It starts out like this (the lines have been truncated):</p>
<pre>
#NEXUS

BEGIN DATA;
	DIMENSIONS NTAX=6 NCHAR=768;
	FORMAT MISSING=? GAP=- DATATYPE=DNA;
	MATRIX
	human      AGAAATATGTCTGATAAAAGAGTTACTTTGATAGAGTAAATAATAGGAGC...
	chimp      AGAAATATGTCTGATAAAAGAATTACTTTGATAGAGTAAATAATAGGAGT...
	bonobo     AGAAATATGTCTGATAAAAGAATTACTTTGATAGAGTAAATAATAGGAGT...
	gorilla    AGAAATATGTCTGATAAAAGAGTTACTTTGATAGAGTAAATAATAGAGGT...
	orangutan  AGAAATATGTCTGACAAAAGAGTTACTTTGATAGAGTAAAAAATAGAGGT...
	siamang    AGAAATACGTCTGACGAAAGAGTTACTTTGATAGAGTAAATAACAGGGGT...
	;
END;
</pre><p>Once loaded, the list of taxa and the actual aligment will be displayed in the main window:</p>
<p><a class="image" href="https://web.archive.org/web/20121204042736/http://beast.bio.ed.ac.uk/File:Tutorial2Beauti1.png" title="tutorial2Beauti1.png"><img alt="tutorial2Beauti1.png" src="https://web.archive.org/web/20121204042736im_/http://beast.bio.ed.ac.uk/w/images/1/1c/Tutorial2Beauti1.png" /></a></p>
<p><a id="Setting_the_dates_of_the_taxa" name="Setting_the_dates_of_the_taxa"></a></p>
<h3>Setting the dates of the taxa</h3>
<p>By default all the taxa are assumed to have a date of zero. This will be correct if, as in this case, all the sequences were sampled at approximately the same point in time (at least on the time scale of their evolutionary history).</p>
<p><a id="Setting_the_evolutionary_model" name="Setting_the_evolutionary_model"></a></p>
<h2>Setting the evolutionary model</h2>
<p>The next thing to do is to click on the Model tab at the top of the main window. This will reveal the evolutionary model settings for BEAST. Exactly which options appear depend on whether the data are nucleotides or amino acids (or nucleotides translated into amino acids). The settings that will appear after loading the Primate data set will be as follows:</p>
<p><a class="image" href="https://web.archive.org/web/20121204042736/http://beast.bio.ed.ac.uk/File:Tutorial2Beauti2.png" title="tutorial2Beauti2.png"><img alt="tutorial2Beauti2.png" src="https://web.archive.org/web/20121204042736im_/http://beast.bio.ed.ac.uk/w/images/7/7a/Tutorial2Beauti2.png" /></a></p>
<p>This tutorial assumes you are familiar with the evolutionary models available, however there are a couple of points to note:</p>
<ol><li>Selecting the Partition into codon positions option assumes that the data are aligned as codons. In this case the data are from mitochondrial tRNAs so this option is inappropriate.</li>
<li>In order to estimate dates of divergence, a previously estimated substitution rate can be specified.</li>
</ol><p><a id="Priors" name="Priors"></a></p>
<h2>Priors</h2>
<p>The next tab allows priors to be specified for each parameter in the model:</p>
<p><a class="image" href="https://web.archive.org/web/20121204042736/http://beast.bio.ed.ac.uk/File:Tutorial2Beauti3.png" title="tutorial2Beauti3.png"><img alt="tutorial2Beauti3.png" src="https://web.archive.org/web/20121204042736im_/http://beast.bio.ed.ac.uk/w/images/0/08/Tutorial2Beauti3.png" /></a></p>
<p>In this tutorial, the default options will be used.</p>
<p><a id="MCMC_operators" name="MCMC_operators"></a></p>
<h2>MCMC operators</h2>
<p>The next stage is to look at the operators for the MCMC. To do this select the Operators tab at the top of the main window. For the Primate dataset, with the model set up as shown in the screen shot above, you will see the following table:</p>
<p><a class="image" href="https://web.archive.org/web/20121204042736/http://beast.bio.ed.ac.uk/File:Tutorial2Beauti4.png" title="tutorial2Beauti4.png"><img alt="tutorial2Beauti4.png" src="https://web.archive.org/web/20121204042736im_/http://beast.bio.ed.ac.uk/w/images/3/3a/Tutorial2Beauti4.png" /></a></p>
<p>Each parameter in the model has one or more "operators". The operators specify how the parameter changes as the MCMC runs. This table lists the parameters, their operators and the tuning settings for these operators. In the first column are the parameter names. These will be called things like hky1.kappa which means the <em>HKY</em> model's <em>kappa</em> parameter (the transition-transversion bias). The next column has the type of operators that are acting on each parameter. For example, the <em>scale operator</em> scales the parameter up or down by a proportion, the <em>random walk operator</em> adds or subtracts an amount to the parameter and the _uniform operator_ simply picks a new value uniformally within a range. Some parameters relate to the tree or to the heights of the nodes of the tree and these have special operators.</p>
<p>The next column, labelled Initial, specifies the starting value for the parameter. Giving a reasonable guess for the parameters may help the MCMC reach equilibrium faster and trying different initial values for multiple runs can be useful for checking for convergence. The next column, labelled Tuning, gives a tuning setting to the operator. Some operators don't have any tuning settings so have n/a under this column. Changing the tuning setting will set how large a move that operator will make which will affect how often that change is accepted by the MCMC which will affect the efficency of the analysis. At the top of the window is an option called Auto Optimize which, when selected, will automatically adjust the tuning setting as the MCMC runs to try to achieve maximum efficiency. At the end of the run a table of the operators, their performance and the final values of these tuning settings will be displayed. These can then be used to set the starting tuning settings in order to minimize the amount of time taken to reach optimum performance in subsequent runs.</p>
<p>The next column, labelled Weight, specifies how often each operator is applied relative to each other. Some parameters have very little interaction with the rest of the model and as a result tend to be estimated very efficiently - an example is the <em>kappa</em>parameter - these parameters can have their operators down-weighted so that they are not changed as often.</p>
<p><a id="Setting_the_MCMC_options" name="Setting_the_MCMC_options"></a></p>
<h2>Setting the MCMC options</h2>
<p>The next tab, MCMC, provides settings to control the MCMC:</p>
<p><a class="image" href="https://web.archive.org/web/20121204042736/http://beast.bio.ed.ac.uk/File:Tutorial2Beauti5.png" title="tutorial2Beauti5.png"><img alt="tutorial2Beauti5.png" src="https://web.archive.org/web/20121204042736im_/http://beast.bio.ed.ac.uk/w/images/d/de/Tutorial2Beauti5.png" /></a></p>
<p>Firstly we have the Length of chain. This is the number of steps the MCMC will make in the chain before finishing. How long this should be depends on the size of the dataset, the complexity of the model and the quality of answer required. The default value of 10,000,000 is entirely arbitrary and should be adjusted according to the size of your dataset. In order examine whether a particular chain length is adequate, the resulting log file can be analysed using <a href="tracer.html" title="Tracer">Tracer</a>. The aim of setting the chain length is to achieve a reasonable <a href="Glossary.html#ESS" title="Glossary">Effective Sample Size (ESS)</a>. Ways of doing this are discussed in <a href="http://sbsweb2-bak.bio.ed.ac.uk/beast/Increasing_ESSs" title="Increasing ESSs">another tutorial</a>.</p>
<p>The next options specify how often the current parameter values should be displayed on the screen and recorded in the log file. The screen output is simply for monitoring the programs progress so can be set to any value (although if set too small, the sheer quantity of information being displayed on the screen may actually slow the program down). For the log file, the value should be set relative to the total length of the chain. Sampling too often will result in very large files with little extra benefit in terms of the precision of the analysis. Sample too infrequently and the log file will not contain much information about the distributions of the parameters. You probably want to aim to store about 10,000 samples so this should be set to the chain length / 10,000.</p>
<p>For this dataset let's initially setting the chain length to 1,000,000 as this will run reasonably quickly on most modern computers.</p>
<p>The final two options give the file names of the log files for the parameters and the trees. These will be set to a default based on the name of the imported NEXUS file but feel free to change these.</p>
<p><a id="Saving_and_Loading_BEAUti_files" name="Saving_and_Loading_BEAUti_files"></a></p>
<h2>Saving and Loading BEAUti files</h2>
<p>If you select the Save option from the File menu this will save a document in BEAUti's own format. Note that although this is an XML document it is not in the format that <a href="BEAST.html" title="BEAST">BEAST</a> understands (it is much simpler and specific to the user interface of BEAUti). The idea is that the settings and data in BEAUti can be saved and loaded at a later time. We suggest you save <a href="BEAUti.html" title="BEAUti">BEAUti</a> files with the extension '.beauti'.</p>
<p><a id="Generating_the_BEAST_XML_file" name="Generating_the_BEAST_XML_file"></a></p>
<h2>Generating the BEAST XML file</h2>
<p>We are now ready to create the BEAST XML file. Select Generate XML... from the File menu and save the file with an appropriate name (we usually end the filename with '.xml'). We are now ready to run the file through <a href="BEAST.html" title="BEAST">BEAST</a>.</p>
<p><a id="Running_BEAST" name="Running_BEAST"></a></p>
<h2>Running BEAST</h2>
<p>Now run BEAST according to the instructions provided (look at the README text file included in the package you downloaded), providing your newly created XML file as input.</p>
<pre>
+-----------------------------------------------\
|            BEAST v1.1.2 2002-2004             |\
| Bayesian Evolutionary Analysis Sampling Trees ||
|     BEAST Library: Build 1.12 2004/08/27      ||
|       Alexei Drummond and Andrew Rambaut      ||
|              University of Oxford             ||
|      http://evolve.zoo.ox.ac.uk/Beast/        ||
\-----------------------------------------------\|
 \-----------------------------------------------\
 +-----------------------------------------------+
 | Components created by:                        |
 |       Alexei Drummond                         |
 |       Roald Forsberg                          |
 |       Oliver Pybus                            |
 |       Andrew Rambaut                          |
 | Thanks to (for use of their code):            |
 |       Korbinian Strimmer                      |
 |       Oliver Pybus                            |
 +-----------------------------------------------+

Read alignment:
  Sequences = 6
      Sites = 768
   Datatype = nucleotide
Site patterns 'patterns' created from positions 1-768 of alignment 'alignment'
  pattern count = 69
TreeLikelihood using native nucleotide likelihood core.

Pre-burnin (10000 states)
0              25             50             75            100
|--------------|--------------|--------------|--------------|
*************************************************************

state	Likelihood  	Root Height 	L(tree)     	L(coalecent)
0	-1,836.8188 	16.9657     	-1,816.8341 	-19.9846
10000	-1,834.4612 	14.8862     	-1,814.4899 	-19.9713
20000	-1,834.4422 	13.6969     	-1,815.1315 	-19.3108
.
.
.
980000	-1,835.2411 	14.1121     	-1,815.4359 	-19.8052
990000	-1,836.8918 	17.5534     	-1,814.5603 	-22.3315
1000000	-1,835.6019 	13.2590     	-1,815.5852 	-20.0167

Operator analysis
Operator                              Pr(accept)  Performance suggestion
hky.kappa                     0.268   0.2583      good
constant.popSize              0.204   0.2645      good
treeModel.rootHeight          0.605   0.2392      good
nodeHeights                           0.3834      slightly high
subtreeSlide                  4.446   0.2559      good
Narrow Exchange                       0.0483      low
Wide Exchange                         0.0182      good
wilsonBalding                         0.0041      low

1.0085 minutes
</pre><p>This took 1 minute to run on a 2GHz G5 Macintosh. Clearly we would be able to run it for much longer. <a href="http://sbsweb2-bak.bio.ed.ac.uk/beast/Analysing_BEAST_output" title="Analysing BEAST output">See the tutorial on analysing MCMC output to see how to improve this estimate</a>.</p>
<p><a id="Estimating_a_divergence_date" name="Estimating_a_divergence_date"></a></p>
<h2>Estimating a divergence date</h2>
<p>The next part of this tutorial involves editing the BEAST XML file in order to estimate the divergence date of a clade in the tree. By default the date of the root of the tree is written to the log file so if this is what we are interested in then we need only use <a href="Tracer.html" title="Tracer">Tracer</a> to analyse the log file and look at the posterior density plot of the root time. If we wish to estimate the time of a particular clade then further steps must be taken. It is assumed that you followed the tutorial up until this point and have created a working BEAST XML file. This should then be opened in a text editor.</p>
<p>The first thing we must do is define the group of taxa that we want to estimate the divergence time of. We will add this new taxa block directly below the one that already exists in the XML file. In this case we are interested in the divergence between humans and chimps so the new taxa block will look like:</p>
<pre>
&lt;taxa id="humanChimp"&gt;
    &lt;taxon idref="human"/&gt;
    &lt;taxon idref="chimp"/&gt;
    &lt;taxon idref="bonobo"/&gt;
&lt;/taxa&gt;
</pre><p>Notice that idref is used rather than id to specify the taxon objects in this taxa block, because we want to refer to the previously defined taxon objects rather than creating new ones. The idref attribute is used to refer to a previously defined element, while id is used to give a unique name to a new element.</p>
<p>The next thing to do is create a tmrcaStatistic that will log the age of the most recent common ancestor (MRCA) of the humanChimp taxa. We do this by adding the following XML element after the treeModel element in the Primates.xml file:</p>
<pre>
&lt;!-- A statistic that returns the time of the most recent common ancestor   --&gt;
&lt;!-- of the given set of taxa.                                              --&gt;
&lt;tmrcaStatistic id="humanChimpTMRCA" name="human/chimp"&gt;
    &lt;mrca&gt;
        &lt;taxa idref="humanChimp"/&gt;
    &lt;/mrca&gt;
    &lt;treeModel idref="treeModel"/&gt;
&lt;/tmrcaStatistic&gt;
</pre><p>We should also add this new statistic to the log so that we can obtain estimates of the mean and 95% HPD intervals after running the MCMC chain. We do this by adding the following line to the file and screen log elements:</p>
<pre>
&lt;tmrcaStatistic idref="humanChimpTMRCA"/&gt;
</pre><p>The &lt;mcmc&gt; element will now look like this:</p>
<pre>
	&lt;mcmc id="mcmc" chainLength="1000000" autoOptimize="true"&gt;
		&lt;compoundLikelihood id="likelihood"&gt;
			&lt;coalescentLikelihood idref="coalescent"/&gt;
			&lt;treeLikelihood idref="treeLikelihood"/&gt;
		&lt;/compoundLikelihood&gt;
		&lt;operators idref="operators"/&gt;
		&lt;log id="screenLog" logEvery="10000"&gt;
			&lt;column label="Likelihood" dp="4" width="12"&gt;
				&lt;compoundLikelihood idref="likelihood"/&gt;
			&lt;/column&gt;
			&lt;column label="Root Height" sf="6" width="12"&gt;
				&lt;parameter idref="treeModel.rootHeight"/&gt;
				&lt;tmrcaStatistic idref="humanChimpTMRCA"/&gt;
			&lt;/column&gt;
			&lt;column label="L(tree)" dp="4" width="12"&gt;
				&lt;treeLikelihood idref="treeLikelihood"/&gt;
			&lt;/column&gt;
			&lt;column label="L(coalecent)" dp="4" width="12"&gt;
				&lt;coalescentLikelihood idref="coalescent"/&gt;
			&lt;/column&gt;
		&lt;/log&gt;
		&lt;log id="fileLog" logEvery="100" fileName="Primates.log"&gt;
			&lt;compoundLikelihood idref="likelihood"/&gt;
			&lt;parameter idref="constant.popSize"/&gt;
			&lt;parameter idref="hky.kappa"/&gt;
			&lt;parameter idref="treeModel.rootHeight"/&gt;
			&lt;tmrcaStatistic idref="humanChimpTMRCA"/&gt;
			&lt;treeLikelihood idref="treeLikelihood"/&gt;
			&lt;coalescentLikelihood idref="coalescent"/&gt;
		&lt;/log&gt;
		&lt;logTree id="treeFileLog" logEvery="100" nexusFormat="true" fileName="Primates.trees"&gt;
			&lt;treeModel idref="treeModel"/&gt;
		&lt;/logTree&gt;
	&lt;/mcmc&gt;
</pre><p>When we run this in <a href="https://web.archive.org/web/20121204042736/http://beast.bio.ed.ac.uk/BEAST" title="BEAST">BEAST</a> as before and when we examine the log file using <a href="https://web.archive.org/web/20121204042736/http://beast.bio.ed.ac.uk/Tracer" title="Tracer">Tracer</a> we obtain the following estimates for the tMRCA of human/chimps:</p>
<p><a class="image" href="https://web.archive.org/web/20121204042736/http://beast.bio.ed.ac.uk/File:Tutorial2Tracer1.png" title="tutorial2Tracer1.png"><img alt="tutorial2Tracer1.png" src="https://web.archive.org/web/20121204042736im_/http://beast.bio.ed.ac.uk/w/images/f/f7/Tutorial2Tracer1.png" /></a></p>
<p>The mean estimate is 5 million years ago with a 95% HPD of (3.54, 6.70).</p>
<p>If there is considerable uncertainty in the phylogenetic relationships of the taxa, it may be desirable to enforce the monophyly of the clade. How to do this is described in <a href="http://sbsweb2-bak.bio.ed.ac.uk/beast/tutorial-5">another tutorial</a>.</p>
<hr /><p><a href="tutorials.html" title="Tutorials">Return to Tutorials</a></p>
</div></div></div>  </div>

  <div class="clearfix">
          <div class="links"></div>
    
      </div>

</div>
          </div>
                          </div></div></div></div> <!-- /.left-corner, /.right-corner, /#squeeze, /#center -->

      
    </div> <!-- /#container -->
  </div> <!-- /#wrapper -->
  </body>
</html>
